File: app/src/main/java/ethiopia/covid/android/ui/adapter/TabAdapter.java

package ethiopia.covid.android.ui.adapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by BrookMG on 5/5/2019 in app.kuwas.android.ui.adapters
 * inside the project Kuwas .
 */
public class TabAdapter extends FragmentStatePagerAdapter {

    private final List<Fragment> mFragments = new ArrayList<>();
    private final List<String> mTabTitles = new ArrayList<>();

    public TabAdapter(@NonNull FragmentManager fm) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    public void addFragment(Fragment fragment, String tabTitle) {
        mFragments.add(fragment);
        mTabTitles.add(tabTitle);
    }

    public List<Fragment> getmFragments() {
        return mFragments;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mTabTitles.get(position);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }
}

File: app/src/main/java/ethiopia/covid/android/ui/adapter/TabAdapter.kt
package ethiopia.covid.android.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

/**
 * Created by BrookMG on 20/6/2020 in ethiopia.covid.android.ui.adapter
 * inside the project CoVidEt .
 */
class TabAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private val mFragments: MutableList<Fragment> = mutableListOf()
    private val mTabTitles: MutableList<String> = mutableListOf()

    fun addFragment(fragment: Fragment, tabTitle: String) {
        mFragments.add(fragment)
        mTabTitles.add(tabTitle)
    }

    fun setFragments(fragments: List<Fragment>) = mFragments.addAll(fragments)

    fun getmFragments(): List<Fragment> = mFragments

    override fun getPageTitle(position: Int): CharSequence? = mTabTitles[position]

    override fun getItem(position: Int): Fragment = mFragments[position]

    override fun getCount(): Int = mFragments.size
}